public/javascripts/Progress/src/MistakeCarousel.js

```diff
@@ -13,12 +13,31 @@ function MistakeCarousel() {
         let labelTypesWithoutData = [];
         labelTypes.forEach((l) => (data[l].length > 0 ? labelTypesWithData : labelTypesWithoutData).push(l));
 
```

Comment:
So here is what I see when I read this code:
```
k = ''
if (a)
    k = 'key1'
 else
    k = 'key2'

x = ''
if (!a && b)
    x = x + f(k) + f('key3')
else
    x = f(k)
```

It's way more complicated and convoluted than it needs to be. All you're doing is building a string based on a few conditions, and it should read like that to others who read your code! I could rewrite the code above like this:
```
x = ''
if (a)
    x = f('key2')
else if (b)
    x = f('key1') + f('key3')
else
    x = f('key1')
```

I'd like to see something that looks more like that!